OsmTrack.py

```diff
@@ -2,26 +2,33 @@
 import numpy as np
 import math
 from PointTag import PointTag
+from TrackLength import TrackLength
+from TrackDifficulty import TrackDifficulty
 from TrackShape import TrackShape
 from geopy.distance import geodesic
 
 CLOSENESS_THRESH_METERS = 200
 LOOP_THRESH_METERS = 100
 SAMPLING_RATIO = 1 / 10
+MID_LENGTH_THRESH = 5  # Tracks who's length is between 20m to 40m are considered as medium-length track.
+LONG_THRESH = 20  # Tracks longer then 40m are considered long.
 
 
 class OsmTrack:
     """
     Holds all the data of a public gps-track collected from osm.
     """
 
-    def __init__(self, segment):
+    def __init__(self, segment, track_id):
+        self.id = track_id
         self.segment = segment
         self.interest_points = set()  # Waterways, historic places, etc...
         self.gps_points = self.extract_gps_points()  # Pandas df (lat, lon, time)
         self.length = self.calculate_length()  # The length of the track (in km)
         self.avg_velocity = self.calculate_avg_velocity()  # The average velocity of the track (in km\h)
```

Comment:
In the current implementation the order of evaluating the members doesn't matter, since the whole object is created before it's  being filtered. 
We might consider to change this design, but I can't think of good alternatives and I don't know if we should mess with it too much (it's done offline, and the real bottleneck is matching interest-points to tracks)